CMakeLists.txt

```diff
@@ -239,6 +250,80 @@ endif()
 
 include(cmake/optional.cmake)
 
+set(hardened_cxx_flags "")
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+  # All versions of gcc that we commonly use for building are subject to bug
+  # https://gcc.gnu.org/bugzilla/show_bug.cgi?id=90348. To work around that, set
+  # -fstack-reuse=none for all gcc builds.
+  try_append_cxx_flags(hardened_cxx_flags "-fstack-reuse=none")
+endif()
+if(HARDENING)
+  if(MSVC)
+    try_append_cxx_flags(core "/DYNAMICBASE")
```

Comment:
Why are these being added to `core` flags, while all other hardening flags are being added to `hardened_cxx_flags`?

Reply:
That code was wrong as those flags belong to a linker, not a compiler. Reworked.